gui/layers/_markers_layer/model.py

```diff
@@ -391,55 +438,66 @@ def get_value(self, position, indices):
             msg = msg + ', ' + self.name + ', index ' + str(value)
         return coord, value, msg
 
-    def add(self, position, indices):
-        """Adds object at given mouse position and set of indices.
-
+    def _add(self, coord):
+        """Adds object at given mouse position
+        and set of indices.
         Parameters
         ----------
-        position : sequence of two int
-            Position of mouse cursor in canvas.
-        indices : sequence of int or slice
-            Indices that make up the slice.
+        coord : sequence of indices to add marker at
         """
-        coord = self._get_coord(position, indices)
         self._size = append(self._size, [np.repeat(10, self.ndim)], axis=0)
         self.data = append(self.data, [coord], axis=0)
         self._selected_markers = len(self.data)-1
 
-    def remove(self, position, indices):
-        """Removes object at given mouse position and set of indices.
-
-        Parameters
-        ----------
-        position : sequence of two int
-            Position of mouse cursor in canvas.
-        indices : sequence of int or slice
-            Indices that make up the slice.
+    def _remove(self):
+        """Removes selected object if any.
         """
-        coord = self._get_coord(position, indices)
         index = self._selected_markers
-        if index is None:
-            pass
-        else:
-
+        if index is not None:
             self._size = delete(self._size, index, axis=0)
             self.data = delete(self.data, index, axis=0)
             self._selected_markers = None
 
-    def move(self, position, indices):
-        """Moves object at given mouse position and set of indices.
+    def _move(self, coord):
```

Comment:
What's your reasoning for making this private?

Reply:
This now only gets called from within the markers layer. It isn't quite generic enough at this point that I'd call if from a notebook, but maybe that will change when we do the shapes stuff and improve this sort of editing / moving api. Maybe we revisit then